buildflow/core/io/gcp/providers/pubsub_subscription.py

```diff
@@ -0,0 +1,91 @@
+from typing import Optional, Type
+
+import pulumi
+import pulumi_gcp
+
+from buildflow.core.credentials import GCPCredentials
+from buildflow.core.io.gcp.strategies.pubsub_strategies import (
+    GCPPubSubSubscriptionSource,
+)
+from buildflow.core.providers.provider import PulumiProvider, SourceProvider
+from buildflow.core.types.gcp_types import GCPProjectID, PubSubSubscriptionName
+from buildflow.io.gcp.pubsub_topic import GCPPubSubTopic
```

Comment:
hit a recursive import error and had to separate Topic & Subscription. We could avoid this by passing the GCPPubSubTopicProvider in this class, similar to what you did with the GCSFileStream provider

Reply:
Yea I think I have a slight preference for that. So only primitives have dependencies on primitives, provider have dependencies on providers, and strategies have dependencies on strategies.